akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -303,7 +303,7 @@ import scala.collection.JavaConverters._
         }
       } else {
         if (activeSubstreamsMap.size == maxSubstreams)
-          fail(new IllegalStateException(s"Cannot open substream for key '$key': too many substreams open"))
+          throw new IllegalStateException(s"Cannot open substream for key '$key': too many substreams open")
```

Comment:
This would be rather inefficient. Perhaps define a separate exception class `with NoStackTrace`. Must be public, in `akka.stream`. Then new instance and throw is not needed each time.

Reply:
Sorry - I got distracted but will try and get this done over the weekend. What’s a good file to declare the exception in, or are you suggesting a new file?

Reply:
Thanks, I think we have a few similar exceptions in akka.stream.
Feels like overkill with one fle per exception but since we don’t have a structure for that, let’s cont like that. We can always move that around later.

Reply:
I'm going to add `NoStackTrace` to the exception and declare it in a companion object. I've not found a package object or other place where exceptions are commonly declared. Besides, I would think it better to keep the exception declared to what causes it, right? Anyhow, changes pushed shortly. 'hope that you like them.